File: Application/src/main/java/com/shareyourproxy/api/rx/event/NotificationCardActionEvent.java
package com.shareyourproxy.api.rx.event;

import com.shareyourproxy.app.adapter.BaseRecyclerViewAdapter;
import com.shareyourproxy.app.adapter.BaseViewHolder;
import com.shareyourproxy.widget.DismissibleNotificationCard.NotificationCard;

/**
 * Created by Evan on 11/10/15.
 */
public class NotificationCardActionEvent {
    public final NotificationCard cardType;
    public final boolean isHeader;
    public final BaseViewHolder holder;
    private final BaseRecyclerViewAdapter adapter;

    public NotificationCardActionEvent(
        BaseRecyclerViewAdapter adapter, BaseViewHolder holder, NotificationCard cardType,
        boolean isHeader) {
        this.adapter = adapter;
        this.holder = holder;
        this.cardType = cardType;
        this.isHeader = isHeader;
    }
}

File: Application/src/main/kotlin/com/shareyourproxy/api/rx/event/NotificationCardActionEvent.kt
package com.shareyourproxy.api.rx.event

import com.shareyourproxy.app.adapter.BaseRecyclerViewAdapter
import com.shareyourproxy.app.adapter.BaseViewHolder
import com.shareyourproxy.widget.DismissibleNotificationCard.NotificationCard

/**
 * Notification card was clicked.
 */
internal class NotificationCardActionEvent(private val adapter: BaseRecyclerViewAdapter, val holder: BaseViewHolder, val cardType: NotificationCard, val isHeader: Boolean)
